docs/pbc/all/miscellaneous/202407.0/tutorials-and-howtos/how-to-extend-the-log-structure-with-additional-data.md

```diff
@@ -0,0 +1,253 @@
+
+        return $data;
+    }
+
+    /**
+     * @return \Generated\Shared\Transfer\CustomerTransfer|null
+     */
+    protected function findCurrentCustomer(): ?CustomerTransfer
+    {
+        $currentRequest = $this->getFactory()->getRequestStackService()->getCurrentRequest();
+
+        if (!$currentRequest || !$currentRequest->hasSession()) {
+            return null;
+        }
+
+        return $this->findCustomerInRequest($currentRequest);
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\CustomerTransfer $customerTransfer
+     *
+     * @return array<string, mixed>
+     */
+    protected function getCurrentRequestData(CustomerTransfer $customerTransfer): array
+    {
+        $currentRequestData = [];
+
+        $currentRequestData['customer_business_unit_name'] = $customerTransfer->getCompanyUserTransferOrFail()
+            ->getCompanyBusinessUnitOrFail()
+            ->getName();
+
+        return $currentRequestData;
+    }
+
+    /**
+     * @param \Symfony\Component\HttpFoundation\Request $request
+     *
+     * @return \Generated\Shared\Transfer\CustomerTransfer|null
+     */
+    protected function findCustomerInRequest(Request $request): ?CustomerTransfer
+    {
+        return $request->getSession()->get('customer data');
+    }
+}
+```
+
+* Register the newly introduced plugin for the **security** log type for the **Yves** application:
+
+  **src/Pyz/Yves/Log/LogDependencyProvider.php**
+
+```php
+<?php
+
+namespace Pyz\Yves\Log;
+
+use Spryker\Yves\Kernel\Container;
+use Spryker\Yves\Log\LogDependencyProvider as SprykerLogDependencyProvider;
+
+class LogDependencyProvider extends SprykerLogDependencyProvider
+{
+    /**
+     * @return list<\Spryker\Shared\Log\Dependency\Plugin\LogProcessorPluginInterface>
+     */
+    protected function getYvesSecurityAuditLogProcessorPlugins(): array
+    {
+        return [
+            // other plugins
+            new CustomerBusinessUnitProcessorPlugin(),
+        ];
+    }
+}
+```
+
+### Option 2: Pass the data to the specific log context
+
+Pass the data directly to the specific log context as needed:
+
+```php
+<?php
+
+use Generated\Shared\Transfer\AuditLoggerConfigCriteriaTransfer;
+use Spryker\Shared\Log\AuditLoggerTrait;
+
+class AnyClassWhereAuditLoggingIsNeeded
+{
+    use AuditLoggerTrait;
+
+    public function someLogic()
+    {
+        //other logic
+
+        $this->getAuditLogger(
+            (new AuditLoggerConfigCriteriaTransfer())->setChannelName('checkout'),
+        )->info('any action', [
+            'tags' => ['any_checkout_action'],
+            'customer_business_unit_name' => $customerTransfer->getCompanyUserTransferOrFail()
+                ->getCompanyBusinessUnitOrFail()
+                ->getName(),
+        ]);
+
+        //other logic
+    }
+}
+```
+
+## Summary
+
+Now when you log in as a customer with a business unit the corresponding business unit name will be added to the log data.
```

Comment:
Quite artificial example, since providing a link to the customer in the log would've solved this task.

@dunqan overall feature seems to require a lot of customization, since the data storage is not connected to the environment - no cross-linking or similar. I understand, we can add links to the log, but that sounds wrong to me.